dask/array/tests/test_overlap.py

```diff
@@ -702,3 +723,53 @@ def test_trim_boundry(boundary):
     x_overlaped = da.overlap.overlap(x, 2, boundary=boundary)
     x_trimmed = da.overlap.trim_overlap(x_overlaped, 2, boundary=boundary)
     assert np.all(x == x_trimmed)
+
+
+def test_map_overlap_rechunks_array_if_needed():
+    # https://github.com/dask/dask/issues/6597
+    expected = np.arange(11)
+    x = da.from_array(expected, chunks=5)
+    y = x.map_overlap(lambda x: x, depth=2, boundary=0)
+    assert y.chunks == ((5, 4, 2),)
```

Comment:
I think that this more clearly  defines what we care about:

```suggestion
    assert all(c >= 2 for c in y.chunks)
```

That way if someone changes the policy in the future this test doesn't gum up their activity.  We really only care that chunks are at least the size of the depth.  We don't want to be too perscriptive about the exact chunking.  My experience is that being too perscriptive results in developers spending time "fixing up the tests" in future work